Add entry on callbacks

tannerwelsh opened the pull request:
Uses the examples of callbacks as blocks in Sinatra request handlers and an event listener in the browser.

Note: this includes links to the following unwritten entries:
- `/function`

tannerwelsh committed:
Add entry on callbacks

Uses the examples of callbacks as blocks in Sinatra request handlers and
an event listener in the browser

zspencer commented on _posts/2014-01-01-callbacks.md:
```diff
@@ -0,0 +1,39 @@
+further-reading:
+  - computer-program
+---
+Callbacks are [subroutines](/subroutines) that are passed around as arguments with the intention of being executed at some point in the future, usually as part of the response to an [event](/events).
```
Is it worth differentiating subroutines from functions/methods?

zspencer replied:
Also, I try to keep the urls singular (callback.md, subroutine,  event) just for consistency sake. Makes it easier to remember what the URL would be.

zspencer commented:
I would say callbacks are more easily understood outside the context of event driven or non-linear programming. I think the content of this is good, but should be in an entry for "non-linear programming" 

For pure callbacks, I would demonstrate `map` and demonstrate how it calls its callback with each element of a collection.

This keeps the definition smaller (no need to understand dom or events) and is still accurate/precise.

A further reading link to non-linear programming would be useful.

tannerwelsh committed:
Change entry title callbacks to callback (singular)

tannerwelsh committed:
Edit entry on callbacks to focus on smaller examples

Omit references to events or nonlinear programming. Place priority
focus on callbacks as functions passed to other functions, using
iterative functions as an example.

Expand upon definition to show how callbacks can be used for delayed
execution.

tannerwelsh commented:
Hey @zspencer I rewrote this entry to narrow the scope a little. Major changes in b43ded5. Let me know what you think.

FYI, I moved the original copy into a new branch on [event-driven programming](https://github.com/codeunion/glossary.codeunion.io/tree/event-driven-programming). Nonlinear programming is a mathematical concept and probably too broad anyway. This entry could cover the why and how of an evented model.

zspencer commented:
:+1: I like this a lot.

tannerwelsh commented:
Thanks, @zspencer! Merging now.
